<?xml version='1.0'?>
<sdf version="1.6">
  <world name="empty">

    <physics name="1ms" type="ignored">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>10.0</real_time_factor>
    </physics>

    <plugin
      filename="gz-sim-physics-system"
      name="gz::sim::systems::Physics">
    </plugin>
    <plugin
        filename="gz-sim-sensors-system"
        name="gz::sim::systems::Sensors">
        <render_engine>ogre2</render_engine>
    </plugin>
    <plugin
      filename="gz-sim-user-commands-system"
      name="gz::sim::systems::UserCommands">
    </plugin>
    <plugin
      filename="gz-sim-scene-broadcaster-system"
      name="gz::sim::systems::SceneBroadcaster">
    </plugin>

    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>1 1 1 1</diffuse>
      <specular>0.5 0.5 0.5 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name="austin_ground_plane">
      <pose>100 0 0 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>1000 1000</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>50</mu>
              </ode>
              <bullet>
                <friction>1</friction>
                <rolling_friction>0.1</rolling_friction>
              </bullet>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>1000 1000</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>
    <model name="austin_track">
        <static>true</static>
        <!-- <pose>0 0 -9 0 0 0</pose> -->
        <!-- <pose>0 0 -8.9 0 0 0</pose> -->
        <link name="track">
            <visual name='track_vis'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001"/> -->
                    <mesh>
                        <uri>../meshes/austin_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
                <material>
                    <ambient>0.8 0.8 0.8 1</ambient>
                    <diffuse>0.8 0.8 0.8 1</diffuse>
                    <specular>0.8 0.8 0.8 1</specular>
                </material>
            </visual>
            <collision name='track_col'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001" /> -->
                    <mesh>
                        <uri>../meshes/austin_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
            </collision>
        </link>
    </model>

    <model name="budapest_track">
        <pose>200 0 0 0 0 0</pose>
        <static>true</static>
        <!-- <pose>0 0 -9 0 0 0</pose> -->
        <!-- <pose>0 0 -8.9 0 0 0</pose> -->
        <link name="track">
            <visual name='track_vis'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001"/> -->
                    <mesh>
                        <uri>../meshes/budapest_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
                <material>
                    <ambient>0.8 0.8 0.8 1</ambient>
                    <diffuse>0.8 0.8 0.8 1</diffuse>
                    <specular>0.8 0.8 0.8 1</specular>
                </material>
            </visual>
            <collision name='track_col'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001" /> -->
                    <mesh>
                        <uri>../meshes/budapest_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
            </collision>
        </link>
    </model>
    <model name="hockenheim_track">
        <static>true</static>
        <pose>300 0 0 0 0 0</pose>
        <!-- <pose>0 0 -9 0 0 0</pose> -->
        <!-- <pose>0 0 -8.9 0 0 0</pose> -->
        <link name="track">
            <visual name='track_vis'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001"/> -->
                    <mesh>
                        <uri>../meshes/hockenheim_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
                <material>
                    <ambient>0.8 0.8 0.8 1</ambient>
                    <diffuse>0.8 0.8 0.8 1</diffuse>
                    <specular>0.8 0.8 0.8 1</specular>
                </material>
            </visual>
            <collision name='track_col'>
                <geometry>
                    <!-- <mesh filename="package://environments/meshes/track.dae" scale="0.001 0.001 0.001" /> -->
                    <mesh>
                        <uri>../meshes/hockenheim_track.stl</uri>
                        <scale>0.001 0.001 0.001</scale>
                    </mesh>
                </geometry>
            </collision>
        </link>
    </model>
  </world>
</sdf>
